src/app/components/album-details/album-details.service.js

```diff
@@ -0,0 +1,24 @@
+export class AlbumDetailsService {
+  constructor($http) {
+    'ngInject';
+    this.$http = $http;
+  }
+  
+  getAlbum(id) {
+    return this.$http.get(`/album/${id}`)
+            .then(response => response.data)
+            .catch(error => { throw error; });
```

Comment:
Technically is not needed because you're returning a promise, and the promise has only two states, if i only return this: `return this.$http.get('/album/${id}').then(response => response.data)` it's still a promise chain, so, in the controller i still can know if a rejection happened, also, is more recommended to use the promise syntax that is `.then(functionIfResolve, functionIfReject)` but it can catch whatever i want about this without having to continue throwing the error, if i want to create another promise based on this yes, but it's a difference, this is promise chaining, the other one is creating a new promise with `$q`